src/playlist-plugin.js

```diff
@@ -0,0 +1,269 @@
+import videojs from 'video.js';
+import Playlist from './playlist.js';
+import AutoAdvance from './auto-advance.js';
+import { isIndexInBounds, log } from './utils.js';
+
+const Plugin = videojs.getPlugin('plugin');
+
+export default class PlaylistPlugin extends Plugin {
+  /**
+   * Creates a new Playlist instance from an array of items.
+   *
+   * @param {Object[]} items - The array of playlist items.
+   * @return {Playlist} The created Playlist instance.
+   */
+  static createPlaylistFrom(items) {
+    return Playlist.from(items, { onError: log.error, onWarn: log.warn });
+  }
+
+  /**
+   * Constructs a PlaylistPlugin instance.
+   *
+   * @param {Object} player - The video.js player instance.
+   * @param {Object} options - The plugin options.
+   */
+  constructor(player, options) {
+    super(player);
+
+    this.playlist_ = null;
+    this.autoAdvance_ = null;
+  }
+
+  /**
+   * Loads a playlist and sets up related functionality.
+   *
+   * @param {Playlist} playlist - The playlist to load.
+   */
+  loadPlaylist(playlist) {
+    this.playlist_ = playlist;
```

Comment:
What if we have a previous playlist and previous auto-advance?
I guess we should clear our listeners from the previous one, and clear previous autoadvance